test/browser-tests/precache/precache.js

```diff
@@ -0,0 +1,175 @@
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// This is a test and we want descriptions to be useful, if this
+// breaks the max-length, it's ok.
+
+/* eslint-disable max-len, no-lonely-if */
+/* eslint-env browser, mocha */
+/* global testHelper */
+
+'use strict';
+
+describe('Test precache method', () => {
+  let compareCachedAssets = function(assetList, cachedAssets) {
+    return new Promise((resolve, reject) => {
+      var cachedAssetsKeys = Object.keys(cachedAssets);
+      cachedAssetsKeys.should.have.length(assetList.length);
+
+      for (var i = 0; i < assetList.length; i++) {
+        var key = location.origin + assetList[i];
+        if (typeof cachedAssets[key] === 'undefined') {
+          reject(new Error('Cache doesn\'t have a cache item for: ' + key));
+        }
+
+        cachedAssets[key].status.should.equal(200);
+      }
+
+      resolve();
+    });
+  };
+
+  var serviceWorkersFolder = '/test/browser-tests/precache/serviceworkers';
+
+  it('should precache all desired assets in precache-valid', done => {
+    var assetList = [
+      '/test/data/files/text.txt',
+      '/test/data/files/text-1.txt'
+    ];
+    testHelper.activateSW(serviceWorkersFolder + '/simple.js')
+    .then(() => {
+      return testHelper.getAllCachedAssets('precache-valid');
+    })
+    .then(cachedAssets => {
+      return compareCachedAssets(assetList, cachedAssets);
+    })
+    .then(() => done(), done);
+  });
+
+  it('should precache a single item in precache-valid', done => {
+    var assetList = [
+      '/test/data/files/text.txt'
+    ];
+    testHelper.activateSW(serviceWorkersFolder + '/single-item.js')
+    .then(() => {
+      return testHelper.getAllCachedAssets('precache-valid');
+    })
+    .then(cachedAssets => {
+      return compareCachedAssets(assetList, cachedAssets);
+    })
+    .then(() => done(), done);
+  });
+
+  it('should precache all desired assets from promises in precache-valid', done => {
+    var assetList = [
+      '/test/data/files/text.txt',
+      '/test/data/files/text-1.txt'
+    ];
+    testHelper.activateSW(serviceWorkersFolder + '/promises.js')
+    .then(() => {
+      return testHelper.getAllCachedAssets('precache-valid');
+    })
+    .then(cachedAssets => {
+      return compareCachedAssets(assetList, cachedAssets);
+    })
+    .then(() => done(), done);
+  });
+
+  it.skip('should precache all desired assets from arrays of arrays in precache-valid', done => {
```

Comment:
I see this test and a few others marked with `skip()`. Are they ready for review? If they get checked in without review, they might just fall through the cracks.

If they need to be checked in as part of this PR but aren't being run, could you elaborate with a comment?